### PR TITLE
Fix can not createwith exe which path contain special characters or unicode languages

### PR DESCRIPTION
**Pull request by sonyps5201314**
fix can not use createwith api to start exes which path contain special characters or unicode languages like Chinese

### CODE REVIEW

**Review by bgianfo — CHANGES_REQUESTED** [review]
Thanks for the contribution!

It looks like you might have mistakenly included  other commits from your other PRs here?
Can you pair this down to only the issue you describe in your pull request description?

It looks like only a3e8ba1c29b0ed258d1bff1761b2cf163252c1ef is applicable to this PR?

**Comment by sonyps5201314** [comment]
@bgianfo, I have rebased to only include this commit, and the commented out code have be removed.

File: src/creatwth.cpp [hunk 1]
```diff
@@ -1304,8 +1306,12 @@ BOOL WINAPI DetourProcessViaHelperDllsW(_In_ DWORD dwTargetPid,
         goto Cleanup;
     }
 
-    hr = StringCchPrintfW(szCommand, ARRAYSIZE(szCommand),
-                          L"rundll32.exe \"%hs\",#1", &helper->rDlls[0]);
+	//for east asia languages and so on, like Chinese, print format with "%hs" can not work fine before user call _tsetlocale(LC_ALL,_T(".ACP"));
+    //so we can't use "%hs" in format string, because the dll that contain this code would inject to any process, even not call _tsetlocale(LC_ALL,_T(".ACP")) before
+	WCHAR szDllName[MAX_PATH];
+	MultiByteToWideChar(CP_ACP, 0, &helper->rDlls[0], -1, szDllName, ARRAYSIZE(szDllName));
```
Comment (bgianfo) [inline src/creatwth.cpp:1312]: We need to check the return code of MultiByteToWideChar for error?

File: src/creatwth.cpp [hunk 2]
```diff
@@ -1304,8 +1306,12 @@ BOOL WINAPI DetourProcessViaHelperDllsW(_In_ DWORD dwTargetPid,
         goto Cleanup;
     }
 
-    hr = StringCchPrintfW(szCommand, ARRAYSIZE(szCommand),
-                          L"rundll32.exe \"%hs\",#1", &helper->rDlls[0]);
+	//for east asia languages and so on, like Chinese, print format with "%hs" can not work fine before user call _tsetlocale(LC_ALL,_T(".ACP"));
+    //so we can't use "%hs" in format string, because the dll that contain this code would inject to any process, even not call _tsetlocale(LC_ALL,_T(".ACP")) before
```
Comment (bgianfo) [inline src/creatwth.cpp:1310]: nit pick: Can we fix the comment here and above to be consistent with the comment style with the rest of the code?

Comment (bgianfo) [reply src/creatwth.cpp:1310]: indention of StringChhPrintf seems to be off by one, in both places you changed as well.

Comment (sonyps5201314) [reply src/creatwth.cpp:1310]: The latest commit has resolved it.

File: src/creatwth.cpp (hunk 1 above)
Comment (sonyps5201314) [reply src/creatwth.cpp]: The latest commit has resolved it.

**Review by bgianfo — APPROVED** [review]

**Comment by bgianfo** [comment]
Thanks again for the contribution @sonyps5201314 !

**Pull request merged**